notify/impl.go

```diff
@@ -957,3 +957,10 @@ func (a *loginAuth) Next(fromServer []byte, more bool) ([]byte, error) {
 	}
 	return nil, nil
 }
+
+func truncate(s string, l int) string {
+	if l <= 10 {
+		return s[:l]
+	}
+	return fmt.Sprintf("%s...", s[:l-3])
```

Comment:
I doubt it matters much, as after this we're going to be doing network traffic which will cost a lot more.